hpat/tests/test_series.py

```diff
@@ -1001,79 +1001,269 @@ def test_impl(A):
         df = pd.DataFrame({'A': np.arange(n)})
         self.assertTrue(isinstance(hpat_func(df.A), np.ndarray))
 
-    def test_series_fillna1(self):
-        def test_impl(A):
-            return A.fillna(5.0)
+    @unittest.skipIf(hpat.config.config_pipeline_hpat_default,
+                     'No support of axis argument in old-style Series.fillna() impl')
+    def test_series_fillna_axis1(self):
+        '''Verifies Series.fillna() implementation handles 'index' as axis argument'''
+        def test_impl(S):
+            return S.fillna(5.0, axis='index')
         hpat_func = hpat.jit(test_impl)
 
-        df = pd.DataFrame({'A': [1.0, 2.0, np.nan, 1.0]})
-        pd.testing.assert_series_equal(hpat_func(df.A),
-                                       test_impl(df.A), check_names=False)
+        S = pd.Series([1.0, 2.0, np.nan, 1.0, np.inf])
+        pd.testing.assert_series_equal(hpat_func(S), test_impl(S))
 
-    # test inplace fillna for named numeric series (obtained from DataFrame)
-    def test_series_fillna_inplace1(self):
-        def test_impl(A):
-            A.fillna(5.0, inplace=True)
-            return A
+    @unittest.skipIf(hpat.config.config_pipeline_hpat_default,
+                     'No support of axis argument in old-style Series.fillna() impl')
+    def test_series_fillna_axis2(self):
+        '''Verifies Series.fillna() implementation handles 0 as axis argument'''
+        def test_impl(S):
+            print("Original series:\n", S)
+            return S.fillna(5.0, axis=0)
         hpat_func = hpat.jit(test_impl)
 
-        df = pd.DataFrame({'A': [1.0, 2.0, np.nan, 1.0]})
-        pd.testing.assert_series_equal(hpat_func(df.A),
-                                       test_impl(df.A), check_names=False)
+        S = pd.Series([1.0, 2.0, np.nan, 1.0, np.inf])
+        pd.testing.assert_series_equal(hpat_func(S), test_impl(S))
 
-    def test_series_fillna_str1(self):
-        def test_impl(A):
-            return A.fillna("dd")
+    @unittest.skipIf(hpat.config.config_pipeline_hpat_default,
+                     'No support of axis argument in old-style Series.fillna() impl')
+    def test_series_fillna_axis3(self):
+        '''Verifies Series.fillna() implementation handles correct non-literal axis argument'''
+        def test_impl(S, axis):
+            return S.fillna(5.0, axis=axis)
+        hpat_func = hpat.jit(test_impl)
+
+        S = pd.Series([1.0, 2.0, np.nan, 1.0, np.inf])
+        axis_values = [0, 'index']
+        for value in axis_values:
```

Comment:
Why not merge the lines:
`for axis in [0, 'index']:`

Reply:
Fixed